### ISSUE TITLE
Update/Add Visual Studio Build Tools 2017

### ISSUE THREAD

**Issue by d1gress** [actions/virtual-environments#35]
**Tool information**
- Tool name: Visual Studio Build Tools 2017
- Add or update? Add
- Desired version: Latest (15.9.16)
- Approximate size: 246MB
- If this is an add request:
  - Brief description of tool: Allows msbuild to build .vcxproj files.
  - URL for tool's homepage: https://docs.microsoft.com/en-us/visualstudio/productinfo/2017-redistribution-vs#vs2017-download

**Virtual environments affected**
- [X] macOS 10.14
- [X] Ubuntu 16.04 LTS
- [X] Ubuntu 18.04 LTS
- [ ] Windows Server 2016 R2
- [X] Windows Server 2019

**Can this tool be installed during the build?**
As far as I know, no.

**Are you willing to submit a PR?**
I have 0 clues what I'm doing as is so, unfortunately no.

**Comment by LorenzE** [comment]
This would be great to have. I am trying to build our project against Qt 5.10.1 which pre-built binaries are only available up to VS 2017, see [here].(https://download.qt.io/online/qtsdkrepository/windows_x86/desktop/qt5_5101/)

**Comment by thejoebourneidentity** [comment]
Hey @d1gress , thanks for the suggestion. Do you mind providing a bit more context on what you're trying to do? For example, why do you need the build tools on all the other images when it's available on `windows-2016`

**Comment by alepauly** [comment]
Closing for now, @d1gress please reopen if you still need this and can help us understand the context. Thanks!

**Comment by danielo515** [comment]
I'm trying to build a qt app, and here are the requirements, which list visual studio:
https://docs.nodegui.org/docs/guides/getting-started#setting-up-on-windows

**Comment by hrueger** [comment]
@alepauly 
Hi, I have the same problem. I'm trying to build https://github.com/stream-labs/obs-studio-node with a custom version http://github.com/obsproject/obs-studio . `obs-studio-node` requires Visual Studio 2015 or 2017, so I went with the `windows-2016` actions environment. However, the installed version of the Windows 10 SDK is 10.0.14393.0. OBS on the other hand requires Windows 10 SDK 10.0.19041.0 or newer. 
I tried installing another version of the Windows 10 SDK with ` choco install windows-sdk-10-version-2004-all` but although the installation process finishes successfully, OBS still only detects the old Windows 10 SDK.

So I want to move to the `windows-2019` environment but Visual Studio 2017 is not available there.

Any idea?
